src/x11/xf86Wacom.c

```diff
@@ -523,22 +511,39 @@ void wcmInitAxis(WacomDevicePtr priv, enum WacomAxisType type,
 			index = 4;
 			label = XIGetKnownProperty(AXIS_LABEL_PROP_ABS_THROTTLE);
 			break;
+		case WACOM_AXIS_SCROLL_X:
+			index = 5;
+			label = XIGetKnownProperty(AXIS_LABEL_PROP_REL_HSCROLL);
+			break;
+		case WACOM_AXIS_SCROLL_Y:
+			index = 6;
+			label = XIGetKnownProperty(AXIS_LABEL_PROP_REL_VSCROLL);
+			break;
 		case WACOM_AXIS_WHEEL:
 		case WACOM_AXIS_RING:
-			index = 5;
+			index = 7;
 			label = XIGetKnownProperty(AXIS_LABEL_PROP_ABS_WHEEL);
 			break;
 		case WACOM_AXIS_RING2:
-			index = 6;
+			index = 8;
 			break;
+
 		default:
 			abort();
 	}
 
+
 	InitValuatorAxisStruct(pInfo->dev, index,
-	                       label,
-	                       min, max, res, min_res, max_res,
-	                       Absolute);
+			label,
+			min, max, res, min_res, max_res,
+			Absolute);
+
+	if (type == WACOM_AXIS_SCROLL_X) {
+		SetScrollValuator(pInfo->dev, index, SCROLL_TYPE_HORIZONTAL, 0xffff, 0);
```

Comment:
I don't think that's quite what you want, is it? `0xffff` is the increment,i.e. the amount when we consider a scroll motion a logical wheel click, this value seems a bit high. I'd pick something smaller and scale to that, it's not like we need 32-bit precision for what is usually a few dozen pixels of scrolling.

Reply:
This was originally [your](https://github.com/linuxwacom/xf86-input-wacom/pull/222#issuecomment-1027505491) suggestion (although I'm not sure how concrete of a suggestion it was) and I just picked it because I didn't see any reason why not. The value for this variable is somewhat arbitrary, however it puts an upper limit to the user configurable value of `PanscrollThreshold`, if it is larger than our increment and the delta is sufficiently small (but not zero!) the result will get rounded down to zero. This creates a sort of "dead-zone", that makes it only possible to scroll if reaching a minimum speed. This might also make it outright impossible to scroll on a theoretical tablet with a really high polling rate. I'm not sure, what is the highest practical value for `PanscrollThreshold`, I personally have 150, but I'm don't know what someone else might want to use. From playing around with it, 0xffff, or 65535, feels actually quite reasonable. 0xfff, or 4096, might be also fine, but a `PanscrollThreshold` value of 5000, even if somewhat impractical, might still get some use in some specific use case.

Reply:
huh, indeed, it was my example. Sorry about that :) Let me think about this "loudly": we're dealing with Wacom devices here in device-resolution units (not mm) and they typically have a resolution around 100 or 200 units/mm. The `PanscrollThreshold` defaults to 1300 or 2600 or 13mm of movement according to the man page, IOW moving 13mm on the tablet generates one scroll click.

So let's assume the scroll increment is 0xffff, corresponding to those 13mm. That means we have ~5000 units per mm available as precision, which is... 25 times the current resolution of a device. That does seem somewhat excessive :) On the other hand, it's really just a number so maybe we shouldn't care too much about it. In XWayland the value is simply `1.0` and then everything is scaled into that range. So come to think of it - maybe 1.0 is the best option here, easy to read and scale into.

Having said all that - this value **must** be completely independent of the device units, it must not correspond to device units or anything because that'll just trigger bugs and hidden dependencies on the device. That increment value is simply a coordinate system without specific meaning, it only provides the *granularity* but not the *distance*. 

The *distance* is defined by the `PanscrollThreshold`, so we scale from device coordinates into the increment using that threshold. So let's assume increment is 1.0 then that means your outgoing delta is simply `delta/panscroll_threshold`. The value of the threshold doesn't matter because that is in device units - 1300 for the 13mm, or 5000 if you want a 50mm scroll zone. That should just work, right? I think the rounding down you saw was merely the result of an integer division?

Reply:
> I think the rounding down you saw was merely the result of an integer division?

It is a result of integer division (or type conversion of a float to integer), either way, the whole pipeline for passing values around, including the `wcmAxisSet` function, `WacomAxisData` struct and `convertAxes`, uses integers which means the whole thing would have to be rewritten to use floats in order to be able to pass non-integer values. As you said, it *really is just a number* and isn't worth this change? I'll at least add a comment. There's no real reason for it to be this high, but it's on the safe side and there's no drawback from it being this high if I'm not missing something.